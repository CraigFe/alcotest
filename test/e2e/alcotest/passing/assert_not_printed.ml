(** Reproduction for the issue for which
    https://github.com/mirage/alcotest/pull/215 is an attempted fix. ASSERT
    lines should only be printed if either `--verbose` is set or the tests fail. *)

let () =
  let open Alcotest in
  run "assert-not-printed"
    [
      ( "alpha",
        [
          Alcotest.test_case "tc0" `Quick (fun () ->
              Alcotest.(check unit) "alpha-0 check" () ());
          Alcotest.test_case "tc1" `Quick (fun () ->
              Alcotest.(check unit) "alpha-1 check" () ());
        ] );
      ( "beta",
        [
          Alcotest.test_case "tc0" `Quick (fun () ->
              Alcotest.(check unit) "beta-0 check" () ());
        ] );
    ]
