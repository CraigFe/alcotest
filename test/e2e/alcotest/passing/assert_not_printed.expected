Testing assert-not-printed.
This run has ID `<uuid>`.
 ...                alpha          0   tc0.[OK]                alpha          0   tc0.
 ...                alpha          1   tc1.[OK]                alpha          1   tc1.
 ...                beta           0   tc0.[OK]                beta           0   tc0.
The full test results are available in `<build-context>/_build/_tests/<uuid>`.
Test Successful in <test-duration>s. 3 tests run.
